src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.ImmutableTypeParameterArgumentAnalysis.cs

```diff
@@ -0,0 +1,654 @@
+								};
+							}
+						);
+					}
+				},
+				SymbolKind.Property
+			);
+
+			// Type arguments of containing types of referenced properties for static properties
+			context.RegisterOperationAction(
+				ctx => {
+					var operation = (IPropertyReferenceOperation)ctx.Operation;
+
+					AnalyzeMaybeStaticMemberAccess(
+						ctx,
+						annotationsContext,
+						immutabilityContext,
+						operation.Property,
+						() => operation.Syntax
+					);
+				},
+				OperationKind.PropertyReference
+			);
+
+			// Type arguments of created objects
+			context.RegisterOperationAction(
+				ctx => {
+					var operation = (IObjectCreationOperation)ctx.Operation;
+
+					SyntaxNodeOrToken getSyntax() => operation.Syntax switch {
+						ImplicitObjectCreationExpressionSyntax implicitObjectCreation => implicitObjectCreation.NewKeyword,
+						ObjectCreationExpressionSyntax objectCreation => objectCreation.Type,
+						_ => operation.Syntax
+					};
+
+					AnalyzeTypeRecursive(
+						ctx.ReportDiagnostic,
+						annotationsContext,
+						immutabilityContext,
+						operation.Type!,
+						getSyntax
+					);
+				},
+				OperationKind.ObjectCreation
+			);
+
+			// Type arguments of defined variables
+			context.RegisterOperationAction(
+				ctx => {
+					var operation = (IVariableDeclarationOperation)ctx.Operation;
+
+					SyntaxNodeOrToken getSyntax() => operation.Syntax switch {
+						VariableDeclarationSyntax variableDeclaration => variableDeclaration.Type,
+						_ => operation.Syntax
+					};
+
+					AnalyzeTypeRecursive(
+						ctx.ReportDiagnostic,
+						annotationsContext,
+						immutabilityContext,
+						operation.GetDeclaredVariables()[ 0 ].Type,
+						getSyntax
+					);
+				},
+				OperationKind.VariableDeclaration
+			);
+
+			// Type arguments of defined variables
+			context.RegisterOperationAction(
+				ctx => {
+					SyntaxNodeOrToken syntax = ctx.Operation.Syntax;
+
+					AnalyzeTypeRecursive(
+						ctx.ReportDiagnostic,
+						annotationsContext,
+						immutabilityContext,
+						ctx.Operation.Type!,
+						() => syntax
+					);
+				},
+				OperationKind.DeclarationExpression,
+				OperationKind.Discard
+			);
+
+			// Type arguments on implemented types
+			context.RegisterSymbolAction(
+				ctx => {
+					var symbol = (INamedTypeSymbol)ctx.Symbol;
+
+					foreach( INamedTypeSymbol @interface in symbol.Interfaces ) {
+						AnalyzeTypeRecursive(
+							ctx.ReportDiagnostic,
+							annotationsContext,
+							immutabilityContext,
+							@interface,
+							() => getSyntax( @interface )
+						);
+					}
+
+					if( symbol.BaseType is not null ) {
+						AnalyzeTypeRecursive(
+							ctx.ReportDiagnostic,
+							annotationsContext,
+							immutabilityContext,
+							symbol.BaseType,
+							() => getSyntax( symbol.BaseType )
+						);
+					}
+
+					SyntaxNodeOrToken getSyntax(
+						INamedTypeSymbol baseTypeOrInterface
+					) {
+						var (Declaration, BaseType) = symbol.ExpensiveGetSyntaxImplementingType(

```

Comment:
nit: lowercase camel for this syntax because they act more like variables in the current scope

Reply:
Yeah, this is what VS spits out, matching the names of the deconstructed tuple. I thought of changing it but opted to not fight it. Can change it.

Reply:
Ahh, it's ok.  just looked awkward